scripts/scil_compute_connectivity.py

```diff
@@ -0,0 +1,382 @@
+             'qb_loops': os.path.join(root_dir, 'qb_loops/'),
+             'pruned': os.path.join(root_dir, 'pruned/'),
+             'no_loops': os.path.join(root_dir, 'no_loops/'),
+             'no_outliers': os.path.join(root_dir, 'no_outliers/')}
+
+    return paths
+
+
+def _get_saving_options(args):
+    saving_options = {'raw': args.save_raw_connections,
+                      'intermediate': args.save_intermediate,
+                      'discarded': args.save_discarded,
+                      'final': True}
+
+    return saving_options
+
+
+def _create_required_output_dirs(out_paths, args):
+    os.mkdir(out_paths['final'])
+
+    if args.save_raw_connections:
+        os.mkdir(out_paths['raw'])
+
+    if args.save_discarded:
+        os.mkdir(out_paths['loops'])
+        os.mkdir(out_paths['outliers'])
+        os.mkdir(out_paths['qb_loops'])
+        os.mkdir(out_paths['removed_length'])
+
+    if args.save_intermediate:
+        os.mkdir(out_paths['pruned'])
+        os.mkdir(out_paths['no_loops'])
+        os.mkdir(out_paths['no_outliers'])
+
+
+def _save_if_needed(streamlines, args, saving_options, out_paths,
+                    save_type, step_type, in_label, out_label):
+    if saving_options[save_type] and len(streamlines):
+        save_from_voxel_space(streamlines, args.labels, args.tracts,
+                              os.path.join(out_paths[step_type],
+                                           '{}_{}.trk'.format(in_label,
+                                                              out_label)))
+
+
+def _symmetrize_con_info(con_info):
+    final_con_info = {}
+    for in_label in list(con_info.keys()):
+        for out_label in list(con_info[in_label].keys()):
+
+            pair_info = con_info[in_label][out_label]
+
+            final_in_label = min(in_label, out_label)
+            final_out_label = max(in_label, out_label)
+
+            if final_con_info.get(final_in_label) is None:
+                final_con_info[final_in_label] = {}
+
+            if final_con_info[final_in_label].get(final_out_label) is None:
+                final_con_info[final_in_label][final_out_label] = []
+
+            final_con_info[final_in_label][final_out_label].extend(pair_info)
+
+    return final_con_info
+
+
+def _prune_segments(segments, min_length, max_length, vox_size):
+    lengths = list(length(segments) * vox_size)
+    valid = []
+    invalid = []
+
+    for s, l in zip(segments, lengths):
+        if min_length <= l <= max_length:
+            valid.append(s)
+        else:
+            invalid.append(s)
+    return valid, invalid
+
+
+def build_args_parser():
+    p = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description=__doc__)
+    p.add_argument('tracks',
+                   help='Path of the tracks file, in a format supported by ' +
+                        'the Nibabel streamlines API.')
+    p.add_argument('labels',
```

Comment:
Since labels is mandatory, if you use dipy.io.streamlines.load_tractogram this could be your reference